### PR TITLE
Fix MSVC compiler warnings

### PR DESCRIPTION
**Pull request by EeroKurimo**
*Description of changes:*

Building the project with Visual Studio 2019 currently fails due to compiler warnings treated as errors, and these fixes should address them.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by EeroKurimo** [comment]
Bummer, a failed test. It worked on my machine!

I don't have access to the failure log, would you able to share it, or maybe even just modify my PR if it's something super trivial?

**Comment by JonathanHenson** [comment]
It’s probably the format check on the clang6 run. Just run clang-format -i <each file you changed> I’ll get you logs as soon as i can log into our aws account

File: tests/IotServiceTest.cpp [hunk 1]
```diff
@@ -12,6 +12,11 @@
  * express or implied. See the License for the specific language governing
  * permissions and limitations under the License.
  */
+
+#ifdef _WIN32
```
Comment (JonathanHenson) [inline tests/IotServiceTest.cpp]: Is this compiler or windows related? My guess is you want to check for msvc not windows....

Comment (EeroKurimo) [reply tests/IotServiceTest.cpp]: It was actually compiler related, so ``_MSC_VER`` would have been the proper define to check.

However I started thinking that ``snprintf`` also satisfies the msvc compiler, which would make the sample identical for all platforms.

**Comment by EeroKurimo** [comment]
The older samples' main.cpp apparently had not been modified since you added the clang-format-6.0 tests, so I applied that to it.

**Comment by EeroKurimo** [comment]
I don't have the rights merge this, so please merge it when makes sense to do it.

**Pull request merged**